toolchain/check/import.cpp

```diff
@@ -258,6 +258,38 @@ static auto CopyAncestorNameScopesFromImportIR(
   return scope_cursor;
 }
 
+static auto LoadImportForOwningFunction(Context& context,
+                                        const SemIR::File& import_sem_ir,
+                                        const SemIR::Function& function,
+                                        SemIR::InstId import_ref) {
+  // Import if function is declared as extern and is a non-owning declaration
+  if (!function.is_extern || !function.non_owning_decl_id.has_value()) {
```

Comment:
The existence of the non-owning declaration will imply behavior, so `is_extern` shouldn't need to be separately checked (if you're hesitant, you can `CARBON_CHECK(function.is_extern)` after the `if` because that should be an invariant). You might consider one of:

1. Checking `extern_library_id` because it's what you're examining below:
```suggestion
  if (!function.extern_library_id.has_value()) {
```

2. Just look at the non owning decl presence:
```suggestion
  if (!function.non_owning_decl_id.has_value()) {
```